core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1122,6 +1122,11 @@ public List<SqlNode> fieldList() {
     }
 
     void addOrderItem(List<SqlNode> orderByList, RelFieldCollation field) {
+      // If the field being ordered on is a Literal value, we can safely skip it.
+      if (!dialect.getConformance().isSortByLiteralAllowed()
+          && field(field.getFieldIndex()) instanceof SqlLiteral) {
+        return;
+      }
```

Comment:
See my comment on the JIRA case. I think a better solution will live in `SqlImplementor.Context.orderField`.